File: tmp/pdfs/fantui_demo_pdf/capture_demo_screens.py
from __future__ import annotations

from pathlib import Path

from playwright.sync_api import sync_playwright


OUT = Path("/Users/Zhuanz/20260407 YJX AI FANTUI LogicMVP/tmp/pdfs/fantui_demo_pdf/screenshots")
VIEWPORT = {"width": 1672, "height": 941}


def safe_click(page, text: str, timeout: int = 2500) -> None:
    try:
        page.get_by_text(text, exact=True).click(timeout=timeout)
    except Exception:
        page.get_by_text(text).first.click(timeout=timeout)


def capture() -> None:
    OUT.mkdir(parents=True, exist_ok=True)
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)

        page = browser.new_page(viewport=VIEWPORT, device_scale_factor=1)
        page.goto("http://127.0.0.1:8002/demo.html", wait_until="domcontentloaded", timeout=15000)
        page.wait_for_timeout(800)
        safe_click(page, "最大反推")
        page.wait_for_timeout(900)
        page.screenshot(path=OUT / "demo-cockpit.png", full_page=False)
        page.close()

        page = browser.new_page(viewport=VIEWPORT, device_scale_factor=1)
        page.goto("http://127.0.0.1:8002/c919_etras_workstation.html", wait_until="domcontentloaded", timeout=15000)
        page.wait_for_timeout(900)
        safe_click(page, "最大反推（MAX REV）")
        page.wait_for_timeout(900)
        page.screenshot(path=OUT / "c919-workstation.png", full_page=False)
        page.close()

        page = browser.new_page(viewport=VIEWPORT, device_scale_factor=1)
        page.goto("http://127.0.0.1:8002/timeline-sim.html", wait_until="domcontentloaded", timeout=15000)
        page.wait_for_timeout(800)
        page.locator("#presetSelect").select_option(label="C919 E-TRAS · TR_Inhibited 阻塞")
        page.locator("#loadPreset").click()
        page.locator("#runBtn").click()
        page.wait_for_timeout(1600)
        page.screenshot(path=OUT / "timeline-sim.png", full_page=False)
        page.close()

        page = browser.new_page(viewport=VIEWPORT, device_scale_factor=1)
        page.goto("http://127.0.0.1:9191/", wait_until="domcontentloaded", timeout=15000)
        page.wait_for_timeout(800)
        safe_click(page, "完全展开")
        page.wait_for_timeout(500)
        try:
            page.locator("#btnChart").click(timeout=1000)
            page.wait_for_timeout(500)
        except Exception:
            pass
        page.screenshot(path=OUT / "c919-panel.png", full_page=False)
        page.close()

        page = browser.new_page(viewport=VIEWPORT, device_scale_factor=1)
        page.goto("http://127.0.0.1:8002/workbench.html", wait_until="domcontentloaded", timeout=15000)
        page.wait_for_timeout(900)
        try:
            safe_click(page, "一键通过验收")
            page.wait_for_timeout(2600)
        except Exception:
            # The workbench still makes a useful artifact screenshot in its initial state.
            page.wait_for_timeout(800)
        page.screenshot(path=OUT / "workbench-bundle.png", full_page=False)
        page.close()

        browser.close()
    print(OUT)


if __name__ == "__main__":
    capture()

File: tmp/slides/fantui_logicmvp_demo_pages/work/capture_demo_screens.mjs
import fs from "node:fs/promises";
import path from "node:path";
import { chromium } from "playwright";

const outDir = "/Users/Zhuanz/20260407 YJX AI FANTUI LogicMVP/tmp/slides/fantui_logicmvp_demo_pages/screens";
await fs.mkdir(outDir, { recursive: true });

const browser = await chromium.launch({ headless: true });
const context = await browser.newContext({
  viewport: { width: 1440, height: 810 },
  deviceScaleFactor: 1,
});

async function shot(page, file) {
  await page.screenshot({ path: path.join(outDir, file), fullPage: false });
}

async function waitQuiet(page) {
  await page.waitForTimeout(800);
}

try {
  const demo = await context.newPage();
  await demo.goto("http://127.0.0.1:8812/demo.html", { waitUntil: "networkidle" });
  await demo.click('button[data-preset="max-reverse"]');
  await waitQuiet(demo);
  await shot(demo, "demo-fantui-cockpit.png");

  const etras = await context.newPage();
  await etras.goto("http://127.0.0.1:8812/c919_etras_workstation.html", { waitUntil: "networkidle" });
  await etras.click('button[data-preset="landing-deploy"]');
  await waitQuiet(etras);
  await shot(etras, "demo-c919-workstation.png");

  const timeline = await context.newPage();
  await timeline.goto("http://127.0.0.1:8812/timeline-sim.html", { waitUntil: "networkidle" });
  await timeline.selectOption("#presetSelect", "fantui_sw1_stuck");
  await timeline.click("#runBtn");
  await timeline.waitForFunction(() => {
    const status = document.querySelector("#status")?.textContent || "";
    const outcome = document.querySelector("#outcomeGrid")?.textContent || "";
    return /完成|OK|成功/i.test(status) || /blocked|FAILED|阻塞|true|false/i.test(outcome);
  }, { timeout: 10000 }).catch(async () => waitQuiet(timeline));
  await shot(timeline, "demo-timeline-sim.png");

  const fanSim = await context.newPage();
  await fanSim.goto("http://127.0.0.1:8812/fan_sim_panel.html", { waitUntil: "networkidle" });
  await waitQuiet(fanSim);
  await shot(fanSim, "demo-monitor-timeline.png");
} finally {
  await browser.close();
}

console.log(`screens=${outDir}`);
